src/Core/src/Layouts/GridLayoutManager.cs

```diff
@@ -974,6 +981,83 @@ static void ExpandStars(double targetSize, double currentSize, Definition[] defs
 				}
 			}
 
+			bool TryExpandStarDefinitionsWithDensityAwareness(Definition[] definitions, double targetSize, double currentSize, double spacing, double starCount, bool limitStarSizes)
+			{
+				// Try to get density from the grid view
+				var density = GetDensity();
+				if (density <= 1.0)
+				{
+					return false; // Skip density-aware logic for density 1.0 to maintain backward compatibility
```

Comment:
You shouldn't need to skip density-aware logic for density 1.0, the same code should be able to run for density 1.0 as well as any density > 1.0

You should be able to just modify the 

```C#
static void ExpandStars(double targetSize, double currentSize, Definition[] defs, double targetStarSize, double starCount)
```

Method and make it density aware and it should then work for cases when the density is 1.0 and when it's greater than 1.0. 

If you don't agree with me, don't force it and just let me know why you don't agree